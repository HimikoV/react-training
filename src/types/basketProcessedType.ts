import { productType } from "./productType";

export type basketProcessedType = productType & {
  quantity: number;
  uid?: string | undefined;
};
