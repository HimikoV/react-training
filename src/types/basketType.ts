export type basketType = {
  id: string;
};
