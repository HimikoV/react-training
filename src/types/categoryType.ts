export type categoryType = {
  categoryName: string;
  categoryId: string;
  key?: string;
};
