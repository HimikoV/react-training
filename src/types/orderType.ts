import { basketType } from "./basketType";

export type orderType = {
  order_id: number;
  items: basketType[];
  date: string;
  delivery_information: {
    name: string;
    surname: string;
    email: string;
    phone: string;
    state: string;
    zipcode: string;
    city: string;
  };
  delivery_option: string;
  payment_option: string;
};
