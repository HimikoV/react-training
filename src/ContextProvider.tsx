import { createContext, useEffect, useState, useMemo } from "react";
import { mapBasketData } from "./mappers/mapBasketData";
import { getDeliveryOptions } from "./service/getDeliveryOptions";
import { getOrderData } from "./service/getOrderData";
import { getPaymentOptions } from "./service/getPaymentOptions";
import { requestData } from "./service/requestData";
import { basketType } from "./types/basketType";
import { finalizeOptionsType } from "./types/finalizeOptionsType";
import { orderType } from "./types/orderType";
import { productType } from "./types/productType";
import { childrenInterface } from "./interfaces/childrenInterface";
import { basketProcessedType } from "./types/basketProcessedType";
import { contextProviderInterface } from "./interfaces/contextProviderInterface";

export const dataContext = createContext<contextProviderInterface>(
  {} as contextProviderInterface
);

const ContextProvider = ({ children }: childrenInterface) => {
  const [basketData, setBasketData] = useState<basketType[]>(
    JSON.parse(localStorage.getItem("basketData") || "[]")
  );
  const [filter, setFilter] = useState<string[]>([]);
  const [data, setData] = useState<productType[]>([]);
  const [loadLimit, setLoadLimit] = useState<number>(9);
  const [basketProcessedData, setBasketProcessedData] = useState<
    basketProcessedType[]
  >([]);
  const [deliveryOptions, setDeliveryOptions] = useState<finalizeOptionsType[]>(
    []
  );
  const [paymentOptions, setPaymentOptions] = useState<finalizeOptionsType[]>(
    []
  );
  const [open, setOpen] = useState<boolean>(false);
  const [orderData, setOrderData] = useState<orderType[]>([]);

  useEffect(() => {
    const fetchData = async () => {
      const responseData = await requestData();
      setData(responseData);
      const responseDelivery = await getDeliveryOptions();
      setDeliveryOptions(responseDelivery);
      const responsePayment = await getPaymentOptions();
      setPaymentOptions(responsePayment);
      const responseOrderData = await getOrderData();
      setOrderData(responseOrderData);
    };
    fetchData();
  }, []);

  useEffect(() => {
    localStorage.setItem("basketData", JSON.stringify(basketData));
  }, [basketData]);

  useEffect(() => {
    mapBasketData(data, basketData, setBasketProcessedData);
  }, [data]);

  const trueValues = useMemo(
    () => ({
      filter,
      setFilter,
      basketData,
      setBasketData,
      data,
      loadLimit,
      setLoadLimit,
      basketProcessedData,
      setBasketProcessedData,
      deliveryOptions,
      setDeliveryOptions,
      paymentOptions,
      setPaymentOptions,
      open,
      setOpen,
      orderData,
      setOrderData,
    }),
    [
      filter,
      setFilter,
      basketData,
      setBasketData,
      data,
      loadLimit,
      setLoadLimit,
      basketProcessedData,
      setBasketProcessedData,
      deliveryOptions,
      setDeliveryOptions,
      paymentOptions,
      setPaymentOptions,
      open,
      setOpen,
      orderData,
      setOrderData,
    ]
  );

  // const values = {};
  return (
    <dataContext.Provider value={trueValues}>{children}</dataContext.Provider>
  );
};
export default ContextProvider;
