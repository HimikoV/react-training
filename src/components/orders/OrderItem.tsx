import { Grid, Link, Box } from "@mui/material";
import { orderType } from "../../types/orderType";

const OrderItem = ({
  orderItem,
  handleOpen,
}: {
  orderItem: orderType;
  handleOpen: (arg: orderType) => void;
}) => {
  return (
    <Grid
      data-testid="orderItem"
      sx={{
        color: "#3d405b",
        justifyContent: "space-between",
        alignItems: "center",
        flexDirection: "row",
        width: "37.5rem",
        minWidth: "31.25rem",
      }}
    >
      <Box
        sx={{
          borderRight: "1px solid black",
          px: "0.5rem",
          py: "0.25rem",
          my: "0.25rem",
        }}
      >
        <Link
          placeholder="buttonLink"
          onClick={() => handleOpen(orderItem)}
          sx={{ cursor: "pointer" }}
        >
          {orderItem.id}
        </Link>
      </Box>
      <Box>
        {orderItem.payment_option} / {orderItem.delivery_option}
      </Box>
      <Box
        sx={{
          borderLeft: "1px solid black",
          px: "0.5rem",
          py: "0.25rem",
          my: "0.25rem",
        }}
      >
        {orderItem.date}
      </Box>
    </Grid>
  );
};

export default OrderItem;
