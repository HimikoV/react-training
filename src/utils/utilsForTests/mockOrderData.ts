import { deliveryInformationType } from "../../types/deliveryInformationType";
import { orderType } from "../../types/orderType";
import { mockBasketData } from "./mockBasketData";

export const mockDeliveryInformation: deliveryInformationType = {
  name: "asdfasd",
  surname: "sdafas",
  email: "asdf@asdfjha.com",
  phone: "48325235235",
  zipcode: "23-325",
  state: "afdas",
  city: "asdfas",
};

export const mockOrderData: orderType[] = [
  {
    id: 1234,
    items: mockBasketData,
    date: "1/1/2022",
    delivery_information: mockDeliveryInformation,
    delivery_option: "DHL",
    payment_option: "cash",
  },
  {
    id: 1235,
    items: mockBasketData,
    date: "1/1/2022",
    delivery_information: mockDeliveryInformation,
    delivery_option: "DHL",
    payment_option: "cash",
  },
];
