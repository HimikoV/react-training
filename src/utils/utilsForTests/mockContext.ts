import {
  mockBasketData,
  mockBasketProcessedDataResult,
} from "./mockBasketData";
import { mockProductData, mockProductDataResult } from "./mockProductData";
import { mockFinalizeData } from "./mockFinalizeData";
import { mockOrderDataArray } from "./mockOrderData";
import { contextProviderInterface } from "../../interfaces/contextProviderInterface";
import { Firestore } from "firebase/firestore";
import { NULL } from "sass";
import { Auth } from "firebase/auth";

export const mockContext: contextProviderInterface = {
  filter: [],
  setFilter: jest.fn(),
  basketData: mockBasketData,
  setBasketData: jest.fn(),
  data: mockProductData,
  loadLimit: 9,
  setLoadLimit: jest.fn(),
  basketProcessedData: mockBasketProcessedDataResult,
  setBasketProcessedData: jest.fn(),
  deliveryOptions: mockFinalizeData,
  setDeliveryOptions: jest.fn(),
  paymentOptions: mockFinalizeData,
  setPaymentOptions: jest.fn(),
  open: false,
  setOpen: jest.fn(),
  orderData: mockOrderDataArray,
  setOrderData: jest.fn(),
  order: false,
  setOrder: jest.fn(),
  sort: "letter",
  setSort: jest.fn(),
  processedData: mockProductDataResult,
  setProcessedData: jest.fn(),
  orderLoadLimit: 20,
  setOrderLoadLimit: jest.fn(),
  firestore: {} as Firestore,
  auth: {} as Auth,
  user: null,
  setUser: jest.fn(),
};
