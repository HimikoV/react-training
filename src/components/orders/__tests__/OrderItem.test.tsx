import renderer from "react-test-renderer";
import OrderItem from "../OrderItem";
import { dataContext } from "../../../ContextProvider";
import { mockContext } from "../../../utils/utilsForTests/mockContext";
import { mockOrderData } from "../../../utils/utilsForTests/mockOrderData";
import { orderType } from "../../../types/orderType";
import { render, fireEvent } from "@testing-library/react";
import { screen } from "@testing-library/dom";
describe("testing OrderItem render", () => {
  const handleOpen = jest.fn();
  const wrapper = (handleOpen: (arg: orderType) => void) => {
    return (
      <dataContext.Provider value={mockContext}>
        <OrderItem orderItem={mockOrderData} handleOpen={handleOpen} />
      </dataContext.Provider>
    );
  };
  it("OrderItem Renders Correctly", () => {
    const orderItem = renderer.create(wrapper(handleOpen)).toJSON();
    expect(orderItem).toMatchSnapshot();
  });
  it("checks if button link clicks", () => {
    render(wrapper(handleOpen));
    const button = screen.getByPlaceholderText("buttonLink");
    fireEvent.click(button);
    expect(handleOpen).toHaveBeenCalled();
  });
});
