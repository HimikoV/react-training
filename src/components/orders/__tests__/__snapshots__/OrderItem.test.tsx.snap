// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing OrderItem render OrderItem Renders Correctly 1`] = `
<div
  className="MuiGrid-root css-47g6wx-MuiGrid-root"
>
  <div
    className="MuiBox-root css-6p0wtr"
  >
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-12gn65i-MuiTypography-root-MuiLink-root"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      placeholder="buttonLink"
    >
      1234
    </a>
  </div>
  <div
    className="MuiBox-root css-0"
  >
    cash
     / 
    DHL
  </div>
  <div
    className="MuiBox-root css-14qsb8q"
  >
    1/1/2022
  </div>
</div>
`;
